File: archive/ios-swiftui/Sources/Theme/Theme.swift
import SwiftUI

// MARK: - Colors
// The warm dark palette established in the process microsites.
// Always dark — preferredColorScheme is locked at the app root.

extension Color {
    static let paper      = Color(red: 0.078, green: 0.067, blue: 0.051)  // #14110d
    static let paperWarm  = Color(red: 0.114, green: 0.094, blue: 0.071)  // #1d1812
    static let paperDeep  = Color(red: 0.067, green: 0.055, blue: 0.039)  // #110e0a
    static let ink        = Color(red: 0.902, green: 0.851, blue: 0.737)  // #e6d9bc
    static let inkSoft    = Color(red: 0.659, green: 0.604, blue: 0.490)  // #a89a7d
    static let inkFaint   = Color(red: 0.431, green: 0.392, blue: 0.318)  // #6e6451
    static let rule       = Color(red: 0.196, green: 0.165, blue: 0.125)  // #322a20
    static let oxblood    = Color(red: 0.788, green: 0.455, blue: 0.408)  // #c97468
}

// MARK: - Fonts

enum Garamond {
    static func regular(_ size: CGFloat) -> Font     { .custom("EBGaramond-Regular",     size: size) }
    static func medium(_ size: CGFloat) -> Font      { .custom("EBGaramond-Medium",      size: size) }
    static func italic(_ size: CGFloat) -> Font      { .custom("EBGaramond-Italic",      size: size) }
    static func mediumItalic(_ size: CGFloat) -> Font { .custom("EBGaramond-MediumItalic", size: size) }
}

enum Mono {
    static func label(_ size: CGFloat = 10, tracking: CGFloat = 2.4) -> Font {
        .system(size: size, weight: .medium, design: .monospaced)
    }
}

File: src/theme.ts
// Ported from Atrium/Sources/Theme/Theme.swift — warm vellum-dark palette,
// EB Garamond serif throughout, monospace for structural labels.

export const colors = {
  paper: "#14110d",
  paperWarm: "#1d1812",
  paperDeep: "#110e0a",
  ink: "#e6d9bc",
  inkSoft: "#a89a7d",
  inkFaint: "#6e6451",
  rule: "#322a20",
  oxblood: "#c97468",
} as const;

// Font family names as registered in app/_layout.tsx via expo-font.
export const fonts = {
  regular: "EBGaramond-Regular",
  medium: "EBGaramond-Medium",
  italic: "EBGaramond-Italic",
  mediumItalic: "EBGaramond-MediumItalic",
} as const;

type TextStyle = {
  fontFamily: string;
  fontSize: number;
  color?: string;
};

// Garamond.<style>(size) → style object, mirroring the Swift helpers.
export const garamond = {
  regular: (size: number): TextStyle => ({ fontFamily: fonts.regular, fontSize: size }),
  medium: (size: number): TextStyle => ({ fontFamily: fonts.medium, fontSize: size }),
  italic: (size: number): TextStyle => ({ fontFamily: fonts.italic, fontSize: size }),
  mediumItalic: (size: number): TextStyle => ({ fontFamily: fonts.mediumItalic, fontSize: size }),
};

// Mono.label(size, tracking) — system monospaced for structural labels.
// `tracking` (points in SwiftUI) maps to letterSpacing here.
export const mono = (size = 10, tracking = 2.4) => ({
  fontFamily: "Courier" as const,
  fontWeight: "500" as const,
  fontSize: size,
  letterSpacing: tracking,
});
